src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -343,6 +343,27 @@ private SelectItem GenerateExpandItem(ExpandTermToken tokenIn)
             PathSegmentToken currentToken = tokenIn.PathToNavigationProp;
 
             IEdmStructuredType currentLevelEntityType = this.EdmType;
+
+            // If we have a previous level binding with the last segment as a Type segment,
+            // The currentLevelEntityType should be the Type of the TypeSegment
+            // E.g /Orders?$expand=Customer/NS.VipCustomer($expand=Trips)
+            // If we are Binding the expanded property Trips, currentLevelEntityType should be the type of the VipCustomer
+            if (this.parsedSegments.Count > 0 && this.parsedSegments.Last() is TypeSegment)
+            {
+                IEdmType typeSegmentType = this.parsedSegments.Last().EdmType;
+
+                IEdmCollectionType collection = typeSegmentType as IEdmCollectionType;
+
+                if (collection != null)
+                {
+                    currentLevelEntityType = (typeSegmentType as IEdmCollectionType).ElementType.Definition as IEdmStructuredType;

```

Comment:
I'm not sure why we need to call `as` twice, couldn't this just read `collection.ElementType.Definition as IEdmStructureType`?

Reply:
#Resolved